### PR TITLE
Develop -> Master

### PR DESCRIPTION
**Pull request by bio-boris**
* Adds actions to run tests
* Adds badging
* changed SHOCK upload in unit tests to DataFileUtil.file_to_shock() as per https://github.com/kbaseapps/ReadsUtils/pull/68/files

### CODE REVIEW

**Comment by MrCreosote** [comment]
Tests are failing

**Comment by MrCreosote** [comment]
In hindsight I'm amazed you were able to get the tests to pass on GHA, given they take > 30m to run. That kind of run time is just bad for business so https://github.com/kbaseapps/ReadsUtils/issues/71

**Comment by MrCreosote** [comment]
```
~/github/kbaseapps/ReadsUtils$ kb-sdk test

Validating module in (/home/[user]/github/kbaseapps/ReadsUtils)

----------------------------------------------------------------------
Ran 113 tests in 1339.045s

OK
*snip*

~/github/kbaseapps/ReadsUtils$ git log
commit 7033ab9b8bb9bcbf5cdf03c1a11de9dcef663a5f (HEAD -> develop, origin/develop)
Merge: b89f88f 6803e6f
Author: [redacted]
Date:   Mon Jan 11 10:42:58 2021 -0600

    Merge pull request #70 from kbaseapps/fixing_tests
    
    Fixing tests
```

File: test/ReadsUtils_server_test.py
Comment (MrCreosote) [inline test/ReadsUtils_server_test.py]: A more accurate name would be `upload_file_to_local_ftp_server` or something like that, but *shrug*

**Comment by MrCreosote** [comment]
For posterity, comments from https://github.com/kbaseapps/ReadsUtils/pull/68 have been addressed here

File: .github/codeql.yml
Comment (MrCreosote) [inline .github/codeql.yml:8]: maybe I'm reading this wrong, but this seems to say that codeql will run daily at 7pm whether there are any code changes or not, and will not run on PRs or pushes? Is that right? If so, is that what we want?

File: .github/codeql.yml
Comment (MrCreosote) [inline .github/codeql.yml:26]: I'm not sure I understand what's going on here. Does github make 2 invisible commits on top of the branch or something?

File: .github/codeql.yml
Comment (MrCreosote) [inline .github/codeql.yml:56]: indenting is a bit odd throughout this file but this particularly sticks out

File: .github/workflows/kb_sdk_test.yaml
Comment (MrCreosote) [inline .github/workflows/kb_sdk_test.yaml:21]: is this a duplicate of the above? If not, what do the top 3 lines do?

**Review by MrCreosote — DISMISSED** [review]
Looks good, just some questions that might result in a few comments for the next person to come along that has to understand this

**Comment by bio-boris** [comment]
@MrCreosote RE: CodeQL, not sure about that, was just copied from https://github.com/kbaseapps/ReadsUtils/pull/67
You may want to direct any questions about that to @jsfillman

File: .github/codeql.yml
Comment (jsfillman) [reply .github/codeql.yml:8]: This is the default setting. I believe it does a scheduled scan as well as a quick diff vs the commit being built. I can look into the specifics more, but this is the stock setting, and is safe to leave as is for now.

File: .github/codeql.yml
Comment (jsfillman) [reply .github/codeql.yml:26]: As mentioned above, this is mostly a stock config. We can dig into the internals a bit more, but there's no risk leaving as-is for the release.

File: .github/codeql.yml
Comment (jsfillman) [reply .github/codeql.yml:56]: Created #72 to review formatting and run through linting.

File: .github/workflows/kb_sdk_test.yaml
Comment (jsfillman) [reply .github/workflows/kb_sdk_test.yaml:21]: It checks out 2 repos: The ReadsUtil code, as well as the custom code the Action itself uses. This is intentional.

**Review by jsfillman — APPROVED** [review]

**Comment by MrCreosote** [comment]
@Yzm1234 I thought we were looking into standardizing on the GHA template?

**Comment by MrCreosote** [comment]
Oh fps, now I see that the file I had most of the questions on was the codeql file, not the sdk file. Ugh.

**Comment by dcchivian** [comment]
@MrCreosote where do things stand with respect to your questions on codeql?  I'd like to close this out if you're happy, and if not, make whatever changes are needed to get you to happy. ( @jsfillman @Yzm1234 @bio-boris )

**Comment by MrCreosote** [comment]
Basically no change. I think the next person that comes along and has to deal with those files will be as confused as I am, and whoever understands the files should take 5mins and put in a few comments explaining what's going on and why they're set up that way, or remove the confusing parts if they can be removed.

**Comment by sychan** [comment]
I think the CodeQL stuff is within the Github actions folder, and people can read up the documentation there.

**Review by sychan — APPROVED** [review]
I think the remaining issues are either minor points about style/naming, or else a matter of reading the documentation for some GHA plumbing. I think this is good to go.

**Review by sychan — APPROVED** [review]

**Review by sychan — APPROVED** [review]

**Comment by dcchivian** [comment]
@MrCreosote please unblock and merge.  thanks.

**Comment by sychan** [comment]
@Yzm1234 I suggest just putting a comment in the CodeQL file that links to this URL: https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning

At that point, Gavin can just accept the changes and we move forward.

**Comment by MrCreosote** [comment]
I read the documentation @sychan linked, but unfortunately it leaves me with an additional open question. @jsfillman states that the codeQL file is just the default config, but from the documentation:

```
The default CodeQL analysis workflow uses the pull_request event to trigger a code scan on pull requests targeted against the default branch.
```

Based on the checks, that's not happening, so presumably something is not working correctly here.

The documentation also does not explain the HEAD^2 part of the config file, and I'm sure I won't be the only one confused by that.

I'm amazed that we've actually come to this point - I'm literally asking for maybe 5 lines of comments, which should take anyone who understands what's going on here less than 5 minutes to do. I asked for this when I first reviewed this PR 2 months ago. We've spent more time talking about the status of the PR (which never changed since my first review) than it would've taken to just make the comments.

I'm beginning to think that no one at KBase actually does understand what's going on with the CodeQL config, which is a much bigger problem if we're committing things that no one understands to our repos.

I strongly believe that the issues above need to be addressed before merging, but @sychan has merge rights, so if he chooses to merge anyway that's up to him.

**Pull request merged**
